src/cogent3/core/new_alignment.py

```diff
@@ -0,0 +1,449 @@
+
+    def __repr__(self) -> str:
+        seq = f"{self.seq[:10]}...{self.seq[-5:]}" if self.seq_len > 15 else self.seq
+        return (
+            f"{self.__class__.__name__}(seq={seq!r}, start={self.start}, "
+            f"stop={self.stop}, step={self.step}, offset={self.offset}, "
+            f"seqid={self.seqid!r}, seq_len={self.seq_len})"
+        )
+
+    def to_rich_dict(self):
+        # get the current state
+        data = {"type": get_object_provenance(self), "version": __version__}
+        data["init_args"] = self._get_init_kwargs()
+        # since we will truncate the seq, we don't need start, stop,
+        # step is sufficient
+        data["init_args"]["step"] = self.step
+        if self.is_reversed:
+            adj = self.seq_len + 1
+            start, stop = self.stop + adj, self.start + adj
+        else:
+            start, stop = self.start, self.stop
+
+        data["init_args"]["seq"] = self.seq[start:stop]
+        data["init_args"]["offset"] = int(self.parent_start)
+        return data
+
+    @classmethod
+    def from_rich_dict(cls, data: dict):
+        init_args = data.pop("init_args")
+        if "offset" in data:
+            init_args["offset"] = data.pop("offset")
+        return cls(**init_args)
+
+    def copy(self, sliced: bool = False):
+        """returns copy
+
+        Parameters
+        ----------
+        sliced
+            if True, the underlying sequence is truncated and the start/stop
+            adjusted
+        """
+        if not sliced:
+            return self.__class__(
+                seq=self.seq,
+                start=self.start,
+                stop=self.stop,
+                step=self.step,
+                offset=self.offset,
+                seqid=self.seqid,
+                seq_len=self.seq_len,
+            )
+        return self.from_rich_dict(self.to_rich_dict())
+
+
+@dataclass
+class SeqData:
+    __slots__ = ("_data", "_alphabet", "_name_order", "_seq_maker")
+
+    def __init__(
+        self,
+        data: dict[str, SeqTypes],
+        alphabet: CharAlphabet,
+        name_order: Optional[Union[tuple[str], None]] = None,
+        seq_maker: Optional[type] = None,
+    ):
+        self._alphabet = alphabet
+        self._name_order = process_name_order(data, name_order)
+        self._seq_maker = seq_maker
+        # Convert to moltype alphabet indicies
+        self._data: dict[str, numpy.ndarray] = {
+            k: seq_index(v, self._alphabet) for k, v in data.items()
+        }
+
+    @property
+    def set_seq_maker(self) -> type:
+        return self._seq_maker
+
+    @set_seq_maker.setter
+    def set_seq_maker(self, value: tuple[type, CharAlphabet]) -> type:
+        seq_maker, alphabet = value
+        if alphabet != self._alphabet:
+            raise ValueError("Alphabets must be the same")
+        assert alphabet == self._alphabet
+        self._seq_maker = seq_maker
+
+    def _make_seq(self, seqid) -> Union[SeqDataView, type]:
```

Comment:
The code in this method should be in the `__getitem__` method for the `str` type